packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,174 @@ it, but it does not seem to be justified. For example:
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+Endpoint supports WebSockets, by default disables all NGINX buffering and all
+redirects from services are rewritten in order to support `/service` prefix
+clients are using.
+
+### Intended use
+It is very important to remember that `/service` endpoint was designed to work
+with root Marathon only. Tasks launched by MarathonOnMarathon instances will not
+be reachable via it.
+
+In the case of Mesos frameworks, this becomes a bit more complex - all the frameworks
+are visible in Mesos' `/state-summary` endpoint output so in theory it is possible
+to access a framework launched using MoM or by hand. Unfortunately, Mesos as
+of now does not enforce framework name uniqueness and thus it is possible to
+launch two frameworks with the same name. AR in such case will route in
+deterministic manner. Root Marathon by default enforces unique names for all the
+tasks and frameworks it has under control.
+
+### Operation
+Under the hood, `/service` endpoint follows non-trivial algorithm in order to
+resolve request path into IP + Port pair.
+
+#### Recursive resolving
+There is no way to tell which part of the path is namespace+service
+name and which one is just a resource address. For example, asumming that
+request path is `/foo/bar/baz/picture.jpg` we can have:
+* namespace+name: `foo`, resource address: `/bar/baz/picture.jpg`
+* namespace+name: `foo/bar`, resource address: `/baz/picture.jpg`
+* namespace+name: `foo/bar/baz`, resource address: `/picture.jpg`
+* namespace+name: `foo/bar/baz/picture.jpg`, resource address: `/`
+
+So the idea is to it iteratively try to resolve each "candidate" namespace+name
+and assume that first-match wins.
+
+Assuming that we deployed `foo/bar` Nginx task, we will see two iterations:
+* first one with `foo` candidate namespace+service, which is not present in the
+cluster
+* second one with `foo/bar` candidate namespace+servie name, which results in
+request being routed to correct backend. In case when user is trying to access
+inexistent service, there are going to be four iterations, each one
+unsuccessful. After the forth one the code will give up.
+
+There is of course a limit on the number of iterations that AR makes. Currently,
+it is set to `10`. This means that a service cannot nest more than nine times and
+that everything after the tenth segment is always treated as a resource component.
+
+#### Resolving individual component
+During each iteration described in previous paragraph, AR tries to resolve the
+a component using the data stored in the cache and if necessary - from MesosDNS:
+* first step is to check if there is a root Marathon task that:
+  * has labels set
+  * `DCOS_SERVICE_NAME` label contents is equal to given `namespace+service` string
```

Comment:
This and the following points do not follow from "if there is a root Marathon task that:". With this one, perhaps "has a `DCOS_SERVICE_NAME` label with contents equal to..."